primitive-types/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2015-2017 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Primitive types shared by Substrate and Parity Ethereum.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "impl-serde", not(feature = "std")))]
+compile_error!("Feature \"impl-serde\" requires feature \"std\" to build.");
+
+#[cfg(all(feature = "impl-rlp", not(feature = "std")))]
+compile_error!("Feature \"impl-rlp\" requires feature \"std\" to build.");
+
+#[macro_use]
+extern crate uint;
+
+#[macro_use]
+extern crate fixed_hash;
+
+#[macro_use]
+extern crate crunchy;
+
+#[cfg(feature = "impl-serde")]
+#[macro_use]
+extern crate impl_serde;
+
+#[cfg(feature = "impl-codec")]
+#[macro_use]
+extern crate impl_codec;
+
+#[cfg(feature = "impl-rlp")]
+#[macro_use]
+extern crate impl_rlp;
+
+construct_uint!(U256, 4);
+#[cfg(feature = "impl-serde")] impl_uint_serde!(U256, 4);
+#[cfg(feature = "impl-codec")] impl_uint_codec!(U256, 4);
+#[cfg(feature = "impl-rlp")] impl_uint_rlp!(U256, 4);
+
+impl U256 {
+	/// Multiplies two 256-bit integers to produce full 512-bit integer
+	/// No overflow possible
+	#[inline(always)]
+	pub fn full_mul(self, other: U256) -> U512 {
```

Comment:
There is a `common` feature of `uint` that has this already. That feature also builds&exports `U256` and `U512` which imo is a bit of a wart (the idea is that `uint` should only provide the **tools** to construct the types, not the types themselves).
@Robbepop do you remember the reason we added the `common` module/feature`? I'm guessing it was to make it less of a hassle to migrate to the new `uint` version? 

What is stopping us from changing `uint_full_mul_reg!` to implement the all of `pub fn full_mul()`? Can't we do that and then implement it for all types created with `construct_uint!`? That way we can remove this.

Reply:
The issue is that `full_mul` involves two uint types. But in `uint` crate we can only construct one at a time and have zero information about the other.